.build/config/cassandra_ci.yaml

```diff
@@ -0,0 +1,342 @@
+pipelines:
+  # All jobs in the pre-commit pipeline must run within constraints and pass
+  # before a commit is merged upstream. Committers are expected to validate
+  # and sign off on this if using non-reference CI environments.
+  #
+  # Failure to do so will lead to commits being reverted.
+  - name: pre-commit
+    jobs:
+      - unit
+      - jvm-dtest
+      - jvm-dtest-upgrade
+      - dtest
+      - dtest-large
+      - dtest-upgrade
+      - dtest-upgrade-large
+      - long-test
+      - cqlsh-test
+
+  # The post-commit pipeline is a larger set of tests that will be matrixed
+  # across other configuration axes (see parameters below) where we expect
+  # different implementations or configurations to only fail rarely if at all.
+  # Failures in tests here will be made visible on JIRA tickets shortly after
+  # test run on reference CI and committers are expected to prioritize
+  # rectifying any failures introduced by their work.
+  - name: post-commit
+    jobs:
+      - unit
+      - cdc
+      - compression
+      - test-oa
+      - test-system-keyspace-directory
+      - test-tries
+      - jvm-dtest
+      - jvm-dtest-upgrade
+      - dtest
+      - dtest-novnode
+      - dtest-offheap
+      - dtest-large
+      - dtest-large-novnode
+      - dtest-upgrade
+      - dtest-upgrade-large
+      - long-test
+      - cqlsh-test
+
+  # These are longer-term, much more rarely changing pieces of infrastructure or
+  # testing. We expect these to fail even more rarely than post-commit.
+  - name: nightly
+    jobs:
+      - stress-test
+      - fqltool-test
+      - test-burn
+
+#-----------------------------------------------------------------------------
+# JOBS
+#
+# Parameters:
+#   job: the name
+#   runtime_env: which runtime_env above to base self off
+#   description: text description of the job
+#   cmd: command to run in shell to execute tests
+#   testlist: command to run in shell to generate full list of tests to run. Expected to output filename value to TESTLIST= env var.
+#   test_timeout: timeout for the test in ms
+#   multiplex: Number of times to multiplex a test of the given suite on addition or change
+#
+# Required env vars:
+#   DIST_DIR=<C*/build dir>
+#   CASSANDRA_DIR=<C* dir>
+#   TESTLIST=<populated by testlist: tag entry on a given job>
+#   JOB=<populated by job: tag entry on a given job>
+#
+# Jobs can replace any base params defined in their resource class, and map
+# types will append new or overwriting keys.
+#
+#-----------------------------------------------------------------------------
+# RUNTIME ENVIRONMENTS
+# i.e. common resource classes
+#-----------------------------------------------------------------------------
+runtime_env:
+  - type: jvm
+    resources: {cpu=4, memory=4g, storage=25g}
+    env:
+      ANT_TEST_OPTS: "-Dno-build-test=true"
+    testlist: find test/unit -name '*Test.java' | sed "s;^test/unit/;;g" | sort
+    test_timeout: 480000
+    multiplex: 500
```

Comment:
what does this mean from an "environment" point of view?  this feels like a config to a specific job and not a generic thing

Reply:
Pulling out of env